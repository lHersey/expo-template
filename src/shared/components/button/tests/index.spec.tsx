import { render } from '@testing-library/react-native';
import React, { ComponentProps, FC } from 'react';
import Providers from 'shared/providers';

import BaseButton from '../index';

const Button: FC<ComponentProps<typeof BaseButton>> = props => {
  return (
    <Providers>
      <BaseButton {...props} />
    </Providers>
  );
};

describe('<Button />', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect.hasAssertions();
      const rendered = render(<Button />).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text', () => {
      expect.hasAssertions();
      const { getByText } = render(<Button />);

      expect(getByText('Hello Button')).toBeTruthy();
    });
  });
});
