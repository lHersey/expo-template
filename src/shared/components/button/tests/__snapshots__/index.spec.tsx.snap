// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshots should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#102A43",
          "fontSize": 24,
        },
      ]
    }
  >
    Hello Button
  </Text>
</View>
`;
